app/controllers/users/two_factor_authentication_setup_controller.rb

```diff
@@ -20,10 +19,16 @@ def create
 
       if result.success?
         process_valid_form
+      elsif result.errors[:selection].include? 'phone'
+        flash[:phone_error] = t('errors.two_factor_auth_setup.must_select_additional_option')
+        redirect_to two_factor_options_path(anchor: 'select_phone')
       else
         @presenter = two_factor_options_presenter
         render :index
       end
+    rescue ActionController::ParameterMissing
+        flash[:error] = t('errors.two_factor_auth_setup.must_select_option')
+        redirect_back(fallback_location: two_factor_options_path, allow_other_host: false)
```

Comment:
Minor: Indentation one too far.

```suggestion
      flash[:error] = t('errors.two_factor_auth_setup.must_select_option')
      redirect_back(fallback_location: two_factor_options_path, allow_other_host: false)
```